.ci/pitest.groovy

```diff
@@ -0,0 +1,448 @@
+            return false
+        }
+        if (mutator() != mutation.mutator()) {
+            return false
+        }
+
+        return sourceFile() == mutation.sourceFile()
+    }
+
+    @Override
+    int hashCode() {
+        int result = sourceFile.hashCode()
+        final int hashCodeMultiplier = 31
+        result = hashCodeMultiplier * result + mutatedClass.hashCode()
+        result = hashCodeMultiplier * result + mutatedMethod.hashCode()
+        result = hashCodeMultiplier * result + mutator.hashCode()
+        result = hashCodeMultiplier * result + description.hashCode()
+        result = hashCodeMultiplier * result + lineContent.hashCode()
+        result = hashCodeMultiplier * result + lineNumber
+        return result
+    }
+
+    @Override
+    String toString() {
+        String toString = """
+            Source File: "${sourceFile()}"
+            Class: "${mutatedClass()}"
+            Method: "${mutatedMethod()}"
+            Line Contents: "${lineContent()}"
+            Mutator: "${mutator()}"
+            Description: "${description()}"
+            """.stripIndent()
+        if (lineNumber() != LINE_NUMBER_NOT_PRESENT_VALUE) {
+            toString += 'Line Number: ' + lineNumber()
+        }
+        return toString
+    }
+
+    @Override
+    int compareTo(Mutation other) {
+        int i = sourceFile() <=> other.sourceFile()
+        if (i != 0) {
+            return i
+        }
+
+        i = mutatedClass() <=> other.mutatedClass()
+        if (i != 0) {
+            return i
+        }
+
+        i = mutatedMethod() <=> other.mutatedMethod()
+        if (i != 0) {
+            return i
+        }
+
+        i = lineContent() <=> other.lineContent()
+        if (i != 0) {
+            return i
+        }
+
+        i = mutator() <=> other.mutator()
+        if (i != 0) {
+            return i
+        }
+
+        if (lineNumber() != LINE_NUMBER_NOT_PRESENT_VALUE
+                && other.lineNumber() != LINE_NUMBER_NOT_PRESENT_VALUE) {
+            i = lineNumber() <=> other.lineNumber()
+            if (i != 0) {
+                return i
+            }
+        }
+
+        return description() <=> other.description()
+    }
+
+    /**
+     * XML format of the mutation. Line number is not included at all as this flag is for
+     * the purpose of directly generating and copying the suppression.
+     *
+     * @return XML format of the mutation
+     */
+    String xmlFormat() {
+        final String xmlOutput = """
+            <mutation unstable="${unstable()}">
+              <sourceFile>${sourceFile()}</sourceFile>
+              <mutatedClass>${mutatedClass()}</mutatedClass>
+              <mutatedMethod>${mutatedMethod()}</mutatedMethod>
+              <mutator>${mutator()}</mutator>
+              <description>${description()}</description>
+              <lineContent>${lineContent()}</lineContent>
+            </mutation>
+            """.stripIndent(10)
+        return xmlOutput
```

Comment:
Can we just return the gstring directly?

Reply:
Done.